### PR TITLE
Updating engineer metrics and removing research metrics

### PR DESCRIPTION
**Pull request by StefanFl**
On the way to remove the significance of the `is_staff` flag, I had a look at metrics:

- Engineer metrics where restricted to staff users. This isn't necessary anymore because every user can import findings, if they have the permissions. Still only superusers can select other users to view, non-superusers can only see their metrics. While I was at it, I gave the user selection dialogue a revamp.
- Research metrics where very specific, only Findings with the test_type `Security Research` are shown here. This should be removed because it seems to be a specific use case for one installation. If it's still needed, I am happy to remove the removal from this PR.
- Found a little bug in the user filter, where authorized users where not removed yet.

### CODE REVIEW

File: dojo/templates/dojo/users.html [hunk 1]
```diff
@@ -53,7 +53,7 @@ <h3 class="has-filters">
                             <th class="nowrap">Phone Number(s)</th>
                             <th class="nowrap, text-center">{% dojo_sort request 'Active' 'is_active' %}</th>
                             <th class="nowrap, text-center">{% dojo_sort request 'Staff' 'is_staff' %}</th>
```
Comment (damiencarol) [inline dojo/templates/dojo/users.html:55]: should we remove this also? notify that it will be deprecated soon?

**Review by damiencarol — APPROVED** [review]

File: dojo/templates/dojo/users.html (hunk 1 above)
Comment (StefanFl) [reply dojo/templates/dojo/users.html:55]: That's for the next release, there are some steps to do until then.

Comment (damiencarol) [reply dojo/templates/dojo/users.html:55]: ok

**Review by valentijnscholten — APPROVED** [review]
I like the cleanup. But for a future PR we could think about letting non-superusers view metrics for other users as well? For example owners and maintainers might want to see these?
It's probably not an wasy change to make as you need to decide which users are shown in the dropdowns. I think JIRA struggles with this and has a global "Browse User" permissions used on most places. But maybe we can think of some nice query based on permissions for the product/producttype.

**Comment by valentijnscholten** [comment]
@devGregA @Maffooch Do you agree with the removal?

**Comment by StefanFl** [comment]
> But for a future PR we could think about letting non-superusers view metrics for other users as well? For example owners and maintainers might want to see these?

Something we can think about in another release.

**Review by kiblik — CHANGES_REQUESTED** [review]

**Comment by devGregA** [comment]
@valentijnscholten will take a good look at this by tomorrow evening (US time).

**Comment by Maffooch** [comment]
I am fine with this. All yours @devGregA

**Review by Maffooch — APPROVED** [review]

File: dojo/metrics/views.py [hunk 2]
```diff
@@ -775,15 +767,14 @@ def engineer_metrics(request):
 
 
 # noinspection DjangoOrm
-@user_passes_test(lambda u: u.is_staff)
 @cache_page(60 * 5)  # cache for 5 minutes
 @vary_on_cookie
 def view_engineer(request, eid):
     user = get_object_or_404(Dojo_User, pk=eid)
     if not (request.user.is_superuser or
             request.user.username == 'root' or
```
[the remark this answers is not in the document]
Comment (StefanFl) [reply dojo/metrics/views.py:775]: Yes, this hard coded user name should be removed. Okay for you @devGregA?

**Comment by devGregA** [comment]
looks good to me

**Review by devGregA — APPROVED** [review]

**Pull request merged**